src/main/java/core/basesyntax/service/impl/WriteReportToFileImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.WriteReportToFile;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class WriteReportToFileImpl implements WriteReportToFile {
+    @Override
+    public void writeReportToFile(Map<String, Integer> fruitReport, String reportFileName) {
+        try (FileWriter writer = new FileWriter(reportFileName)) {
+            writer.write("fruit,quantity\n");
+            for (Map.Entry<String, Integer> entry : fruitReport.entrySet()) {
+                String fruit = entry.getKey();
+                int quantity = entry.getValue();
+                writer.write(fruit + "," + quantity + "\n");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to file: ", e);
+        }
+    }
```

Comment:
lets divide this logic into 2 services
- ReportCreator
- FileWriter
to divide logic, according to Single Responsibility